src/platform/networking/node/chatStream.ts

```diff
@@ -46,7 +115,18 @@ export function prepareChatCompletionForReturn(
 		content: toTextParts(messageContent),
 	};
 
-	sendEngineMessagesTelemetry(telemetryService, [rawMessageToCAPI(message)], telemetryData);
+	// Create enhanced message for telemetry with usage information
+	const telemetryMessage = rawMessageToCAPI(message);
+
+	// Add usage information if available
+	if (c.usage) {
+		(telemetryMessage as any).usage = c.usage;
```

Comment:
I'd prefer that we don't use objects as random property bags but put this in telemetryData or somewhere else properly typed